protocol/x/clob/types/message_batch_cancel.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+)
+
+var _ sdk.Msg = &MsgBatchCancel{}
+
+// NewMsgBatchCancel constructs a MsgBatchCancel.
+func NewMsgBatchCancel(subaccountId types.SubaccountId, cancelBatch []OrderBatch, goodTilBlock uint32) *MsgBatchCancel {
+	return &MsgBatchCancel{
+		SubaccountId:     subaccountId,
+		ShortTermCancels: cancelBatch,
+		GoodTilBlock:     goodTilBlock,
+	}
+}
+
+// ValidateBasic performs stateless validation for the `MsgBatchCancel` msg.
+func (msg *MsgBatchCancel) ValidateBasic() (err error) {
```

Comment:
so ValidateBasic is actually optional as of v0.50 - I think it's still supported for backwards compatibility but can you maybe spend 10-15 minutes to see how it works w/ the new version and what was the motivation? kind of curious about this

Reply:
just wondering if transactions that are obviously wrong (e.g. failing ValidateBasic) can now enter mempool and be included in a block

Reply:
https://github.com/cosmos/cosmos-sdk/issues/15648

Seems like they have moved all validation to the msg server.

Example PRS
https://github.com/cosmos/cosmos-sdk/pull/15832

Based on this comment
https://github.com/cosmos/cosmos-sdk/pull/15782/files#r1163771773
it seems like we lose the functionality of verifying if the proposal message is correct before execution. Apparently simulation covers that case. However, a bit confused as users can still submit messages without simulating first.